File: Data Structures & Algorithms/max-water-container/submission-3.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int val = 0, l = 0, r = heights.size()-1;
        while( l < r){
            int curr = min(heights[r], heights[l]) * (r-l);
            val = max(val, curr);
            if(heights[l] < heights[r]){
                l++;
            }else{
                r--;
            }
        }
        return val;
    }
};

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-3.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        ls, lt = 0, 0
        curr = 0
        while ls < len(s):
            if lt < len(t) and s[ls] == t[lt]:
                curr +=1
                lt +=1
            ls +=1


        return len(t) - curr